src/Fixer/Whitespace/NoExtraWhitespaceFixer.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Remove extra whitespace buried inside of any SLOC.
+ *
+ * @author Tristan Strathearn <r3oath@gmail.com>
+ */
+final class NoExtraWhitespaceFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if ($token->isWhitespace(" \0\x0B")) {
+                $space = $token->getContent();
+
+                // Here we'll normalize all consecutive whitespace into a
+                // single whitespace character.
+                if (strlen($space) > 1) {
+                    $token->setContent(' ');
+                }
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Remove extra whitespace buried inside of any SLOC.';
```

Comment:
This can be more descriptive as not everyone knows what SLOC means, and there is no clear level what is considered "extra" whitespace. What if someone uses multi line if-statements?

https://github.com/rollerworks/search/blob/master/src/ConditionOptimizer/RangeOptimizer.php#L254-L258
https://github.com/rollerworks/search/blob/master/src/ConditionOptimizer/RangeOptimizer.php#L309-L310 (extreme example).